src/dmd/expression.d

```diff
@@ -1818,70 +1819,69 @@ extern (C++) final class IntegerExp : Expression
         return value;
     }
 
-    void setInteger(dinteger_t value)
+    private void setInteger(dinteger_t value)
     {
-        this.value = normalize(type.toBasetype().ty, value);
+        this.value = value;
+        normalize();
     }
 
-    static dinteger_t normalize(TY ty, dinteger_t value)
+    void normalize()
     {
         /* 'Normalize' the value of the integer to be in range of the type
          */
-        dinteger_t result;
-        switch (ty)
+        switch (type.toBasetype().ty)
         {
         case Tbool:
-            result = (value != 0);
+            value = (value != 0);
```

Comment:
This refactor seems largely unrelated.  Also please see #8 here: https://forum.dlang.org/post/p6oibo$1lmi$1@digitalmars.com